sdk/servicebus/service-bus/test/serviceBusClient.spec.ts

```diff
@@ -47,1088 +40,899 @@
-//         TestClientType.PartitionedTopicWithSessions,
-//         TestClientType.PartitionedSubscriptionWithSessions,
-//         "",
-//         true
-//       );
-
-//       await testCreateReceiver(
-//         getOpenReceiverErrorMsg(
-//           ClientType.SubscriptionClient,
-//           receiverClient.entityPath,
-//           TestMessage.sessionId
-//         ),
-//         true
-//       );
-//     });
-//   });
-// });
+describe("Errors with non existing Namespace #RunInBrowser", function(): void {
+  let sbClient: ServiceBusClient;
+  let errorWasThrown: boolean;
+  beforeEach(() => {
+    sbClient = new ServiceBusClient(
+      "Endpoint=sb://a;SharedAccessKeyName=b;SharedAccessKey=c;EntityPath=some-queue"
+    );
+    errorWasThrown = false;
+  });
+  afterEach(() => {
+    return sbClient.close();
+  });
+
+  const testError = (err: Error | MessagingError): void => {
+    const expectedErrCode = isNode ? "ENOTFOUND" : "ServiceCommunicationError";
+    if (!isMessagingError(err)) {
+      should.equal(true, false, "Error expected to be instance of MessagingError");
+    } else {
+      should.equal(err.code, expectedErrCode, "Error code is different than expected");
+      errorWasThrown = true;
+    }
+  };
+
+  it("throws error when sending data to a non existing namespace", async function(): Promise<void> {
+    await sbClient
+      .createSender("some-queue")
+      .send({ body: "hello" })
+      .catch(testError);
+
+    should.equal(errorWasThrown, true, "Error thrown flag must be true");
+  });
+
+  it("throws error when creating batch data to a non existing namespace", async function(): Promise<
+    void
+  > {
+    const sender = sbClient.createSender("some-queue");
+    await sender.createBatch().catch(testError);
+    should.equal(errorWasThrown, true, "Error thrown flag must be true");
+  });
+
+  it("throws error when sending batch data to a non existing namespace", async function(): Promise<
+    void
+  > {
+    const sender = sbClient.createSender("some-queue");
+    await sender.sendBatch(1 as any).catch(testError);
+    should.equal(errorWasThrown, true, "Error thrown flag must be true");
+  });
+
+  it("throws error when receiving batch data to a non existing namespace", async function(): Promise<
+    void
+  > {
+    const receiver = sbClient.createReceiver("some-queue", "peekLock");
+    await receiver.receiveBatch(10).catch(testError);
+
+    should.equal(errorWasThrown, true, "Error thrown flag must be true");
+  });
+
+  it("throws error when receiving streaming data from a non existing namespace", async function(): Promise<
+    void
+  > {
+    const receiver = sbClient.createReceiver("some-queue", "peekLock");
+    receiver.subscribe({
+      async processMessage() {
+        throw "processMessage should not have been called when receive call is made from a non existing namespace";
+      },
+      async processError(err) {
+        testError(err);
+      }
+    });
+
+    await delay(3000);
```

Comment:
In the future perhaps we should make these do a check on 'err' being thrown so we don't have to have these sort of fixed timeouts.

Reply:
Removed the fixed delays with checkWithTimeout.